GoogleCloudPlatform/microservices-demo
# Convert CRLF to LF

### Background 
* This fixes [issue 1347](https://github.com/GoogleCloudPlatform/microservices-demo/issues/1347) — which [pull-request 1688](https://github.com/GoogleCloudPlatform/microservices-demo/pull/1688) didn't fully fix.
* `LF` (invisible character) is used by **Mac/Linux** to denote the end of a line in a file.
* `CRLF` (invisible characters) is used by **Windows** to denote the end of a line in a file.
* What the `.gitattributes` file from [pull-request 1688](https://github.com/GoogleCloudPlatform/microservices-demo/pull/1688) does:
    * it ensures newly committed text files (i.e., not binaries/PNGs/JPEG) use `LF` in remote — not necessarily locally
    * it **does not** convert _existing_ files that use `CRLF` to `LF`
    * it also used `CRLF` locally on Windows and `LF` locally on Mac and Linux
* I used `git ls-files --eol` to check which text files (ignoring png/jpeg/jar/tgz/etc.) were still using `CRLF`. I found that the following files were still using `CRLF`:
    * `src/cartservice/cartservice.sln`
    * `src/adservice/gradlew.bat`
* @marcghanime mentioned [in this comment](https://github.com/GoogleCloudPlatform/microservices-demo/issues/1347#issuecomment-1562847984) that they were able to fix the issue by converting `src\adservice\gradlew.bat` (and 2 other files) to `LF` (from `CRLF`).

### Change Summary
 * I ran: `git add --renormalize .` to convert the following files to `LF` (from `CRLF`):
    * `src/cartservice/cartservice.sln`
    * `src/adservice/gradlew.bat`
* change the `.gitattributes` file such that even local files on Windows use `LF` — not `CRLF`.

### Testing Procedure
* We should ensure:
     * GitHub checks of this PR were able to successfully build the `adservice` Dockerfile 
     * we are able to `docker build` the `adservice` on Windows
